src/test/java/org/example/Operations/CollectionOperationsTest.java

```diff
@@ -1,250 +1,483 @@
-//        Response response = rods.collections().list(token, "/tempZone/home/rods");
-//        CollectionsList mapped = JsonUtil.fromJson(response.getBody(), CollectionsList.class);
-//
-//        entries = mapped.getEntries();
-//    }
-//
-//    private void resetCollections() {
-//        updateList();
-//
-//        // Ensure the list is empty
-//        if (entries != null && !entries.isEmpty()) {
-//            for (String lpath : entries) {
-//                rods.collections().remove(token, lpath);
-//                rods.dataObject().remove(token, lpath, 0);
-//            }
-//        }
-//        updateList();
-//        assertEquals("There should be nothing listed in entries when running .list()", null, entries);
-//    }
-//
-//    private int getIrodsResponseStatusCode(String jsonString) {
-//        ObjectMapper objectMapper = new ObjectMapper();
-//
-//        try {
-//            JsonNode rootNode = objectMapper.readTree(jsonString);
-//
-//            JsonNode statusCodeNode = rootNode.path("irods_response").path("status_code");
-//            return statusCodeNode.asInt();
-//        } catch (Exception e) {
-//            e.printStackTrace();
-//        }
-//        return -1;
-//    }
-//
-//
-//
-//}
\ No newline at end of file
+package org.example.Operations;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.example.Properties.Collection.CollectionsListParams;
+import org.example.Properties.Collection.CollectionsRemoveParams;
+import org.example.Serialize.ModifyMetadataOperations;
+import org.example.Serialize.ModifyPermissionsOperations;
+import org.example.Util.Permission;
+import org.example.Wrapper;
+import org.example.Util.Response;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+public class CollectionOperationsTest {
+    private Wrapper rods;
+    private String token;
+    private Wrapper alice;
+    private String aliceToken;
+    private Response res;
+    private ObjectMapper mapper = new ObjectMapper();
+    @Before
+    public void setup() {
+        String address = "52.91.145.195";
+        String port = "8888";
+        String version = "0.3.0";
+
+        String baseUrl = "http://" + address + ":" + port + "/irods-http-api/" + version;
+
+        // Create client
+        rods = new Wrapper(baseUrl, "rods", "rods");
+        rods.authenticate();
+        token = rods.getAuthToken();
+
+        // Create alice user
+        rods.userGroupOperations().create_user(token, "alice", "tempZone", "rodsuser");
+        rods.userGroupOperations().set_password(token, "alice", "tempZone", "alicepass");
+        alice = new Wrapper(baseUrl, "alice", "alicepass");
+        alice.authenticate();
+        aliceToken = alice.getAuthToken();
+
+        resetPath(rods, token, "/tempZone/home/rods");
```

Comment:
It just ensures that there are no entries in `/tempZone/home/rods`. If I have a test that creates a collection but fails halfway through before it gets a chance to delete the collection then when I try testing again it will automatically fail in creating the collection so I just added the method to give it a clean slate each time.

Reply:
Consider renaming the function to better express what it does. Think about what the next developer who's never seen this may think.

Reply:
can't each test always clean up it's own messes?   rather than having other tests have to 'repave' things?

Reply:
When I was writing and executing the tests, if the test failed midway it doesn't get to the end to delete the collection that was initially created in the test. So when I execute again, the test will fail because it won't create a collection that already exists. Unless I'm misunderstanding what you're asking?

Reply:
right.. is there a 'finally' clause that be used in case of an error or failure?

Reply:
Ohh I completely forgot about 'finally', oops. Then yes, I'll be able to add that to avoid having other outside functions to clean things up

Reply:
excellent.   this is the same approach we use in the python tests in the server.

Reply:
https://github.com/irods/irods/tree/main/scripts/irods/test

Reply:
I think the issue is if I'm, for example, testing to make sure renaming works, using the finally clause is a little trickier because I won't know when it failed of course so I wouldn't know whether to remove the original name or the renamed one unless I just take in the list of entries and remove the pathnames listed, but at that point, I feel like I'm essentially just doing the same thing I was originally doing. Or if it failed immediately and never created the collection in the first place, then it'll again be removing something that doesn't exist? I'm not really sure

Reply:
i think you can issue a remove command with the 'force' flag, and then it doesn't matter if it exists...

the idea is that the finally block guarantees the system is back in the state it started in.

Reply:
Like, I should make a method that forces it to remove? I don't think the remove command has a force flag unless I'm misunderstanding what you're saying.

Reply:
What you want to do is:
- Create a test collection and do the work there
- Track all permission changes so you can restore them at the end
- Use the `recurse` and `no-trash` options to recursively remove the test collection

For example _(the function names may be incorrect, but the idea is there)_:
```java
// All data objects and collections should live under this collection.
// It makes it easier to clean up the test area.
private String sandboxCollection;

void beforeTest()
{
    // Init sandbox collection and other things shared across tests.
}

void afterTest()
{
    // Clean up things shared across tests.
}

@Test
void testTheThing()
{
    // Init things specific to this test.

    try {
        // Create data objects, etc inside sandboxCollection.
    }
    finally {
        // Clean up things specific to this test.
        // - Restore appropriate permissions
        // - Permanently remove data objects and collections created by this test
    }
}
```

Reply:
https://github.com/irods/irods_client_http_api/blob/main/API.md#remove has a no-trash option... that is same as 'force'... well, i expect it will work even if it doesn't exist... maybe that assumption DOES need to be tested.

Reply:
Yea I just tried removing a collection that doesn't exist with both `no-trash` and `recurse` set to 1 but it still gives the "not a collection" error

Reply:
It's fine if attempting to clean up a non-existent collection results in an error. Your test just doesn't need to assert anything on it.

Reply:
Done.